### PR TITLE
[OPIK-418] Add endpoint to return all feedback score names

### PR DESCRIPTION
**Pull request by thiagohora**
## Details
- Add endpoint to retrieve all feedback score names
- Fix parallel container starting

## Issues
OPIK-418

## Testing
- Filter by `project_id`
- Filter by `with_experiments_only`
- Both filters
- No filters

### CODE REVIEW

File: apps/opik-backend/src/main/java/com/comet/opik/domain/FeedbackScoreDAO.java [hunk 1]
```diff
@@ -153,6 +155,47 @@ AND entity_id IN (
             ;
             """;
 
+    private static final String SELECT_FEEDBACK_SCORE_NAMES = """
+            SELECT
+                distinct name
+            FROM (
+                SELECT
```
Comment (idoberko2) [inline apps/opik-backend/src/main/java/com/comet/opik/domain/FeedbackScoreDAO.java:162]: Seems like the nested query here is unnecessary if the inner query is added with `distinct`. Unless I'm missing something.

File: apps/opik-backend/src/main/java/com/comet/opik/domain/FeedbackScoreDAO.java [hunk 2]
```diff
@@ -318,6 +361,42 @@ public Mono<Void> deleteByEntityIds(
...
+    }
+
+    private void bindTemplateParam(UUID projectId, boolean withExperimentsOnly, ST template) {
+        if (projectId != null) {
+            template.add("project_id", projectId);
+        }
+
+        if (withExperimentsOnly) {
+            template.add("with_experiments_only", withExperimentsOnly);
+        }
```
Comment (idoberko2) [inline apps/opik-backend/src/main/java/com/comet/opik/domain/FeedbackScoreDAO.java]:
```suggestion
        template.add("with_experiments_only", withExperimentsOnly);
```

File: apps/opik-backend/src/test/java/com/comet/opik/api/resources/v1/priv/FeedbackScoreResourceTest.java [hunk 3]
```diff
@@ -0,0 +1,264 @@
...
+
+            fetchAndAssertResponse(userProjectId, withExperimentsOnly, names, otherNames, projectId, apiKey,
+                    workspaceName);
+        }
+
+        Stream<Arguments> getFeedbackScoreNames__whenGetFeedbackScoreNames__thenReturnFeedbackScoreNames() {
+            return Stream.of(
+                    Arguments.of(true, false,
+                            PodamFactoryUtils.manufacturePojoList(podamFactory, String.class),
+                            PodamFactoryUtils.manufacturePojoList(podamFactory, String.class)),
```
Comment (idoberko2) [inline apps/opik-backend/src/test/java/com/comet/opik/api/resources/v1/priv/FeedbackScoreResourceTest.java]: I don't see a reason why these should be passed as arguments. Can we move them to the test body?

File: apps/opik-backend/src/test/java/com/comet/opik/api/resources/v1/priv/FeedbackScoreResourceTest.java [hunk 4]
```diff
@@ -0,0 +1,264 @@
...
+                : Stream.of(names, otherNames).flatMap(List::stream).toList();
+
+        try (var actualResponse = webTarget
+                .request()
+                .header(HttpHeaders.AUTHORIZATION, apiKey)
+                .header(WORKSPACE_HEADER, workspaceName)
+                .get()) {
+
+            // then
+            assertThat(actualResponse.getStatusInfo().getStatusCode()).isEqualTo(200);
```
Comment (idoberko2) [inline apps/opik-backend/src/test/java/com/comet/opik/api/resources/v1/priv/FeedbackScoreResourceTest.java]: Minor
```suggestion
            assertThat(actualResponse.getStatusInfo().getStatusCode()).isEqualTo(HttpStatus.SC_OK);
```

**Review by idoberko2 — COMMENTED** [review]
Looks great! I really liked the test clients.
Had a few comments.

File: apps/opik-backend/src/main/java/com/comet/opik/domain/FeedbackScoreDAO.java (hunk 1 above)
Comment (thiagohora) [reply apps/opik-backend/src/main/java/com/comet/opik/domain/FeedbackScoreDAO.java:162]: The issue here is that rows can be duplicated in Clickhouse. When there are updates, Clickhouse takes some time to remove the previous "version of the row". If you check, we always do this order by and limit by id

Comment (idoberko2) [reply apps/opik-backend/src/main/java/com/comet/opik/domain/FeedbackScoreDAO.java:162]: Makes sense

File: apps/opik-backend/src/main/java/com/comet/opik/domain/FeedbackScoreDAO.java (hunk 2 above)
Comment (thiagohora) [reply apps/opik-backend/src/main/java/com/comet/opik/domain/FeedbackScoreDAO.java]: The template doesn't distinguish false from null, so in this case, we only want to render that part of the query if this variable is true. If it is false, we should remove the clause of the query.

File: apps/opik-backend/src/test/java/com/comet/opik/api/resources/v1/priv/FeedbackScoreResourceTest.java (hunk 3 above)
Comment (thiagohora) [reply apps/opik-backend/src/test/java/com/comet/opik/api/resources/v1/priv/FeedbackScoreResourceTest.java]: Good point!

File: apps/opik-backend/src/main/java/com/comet/opik/domain/FeedbackScoreDAO.java (hunk 2 above)
Comment (idoberko2) [reply apps/opik-backend/src/main/java/com/comet/opik/domain/FeedbackScoreDAO.java]: Correct, but `withExperimentsOnly` is of type `boolean` and therefore cannot be `null` so there is no difference

Comment (thiagohora) [reply apps/opik-backend/src/main/java/com/comet/opik/domain/FeedbackScoreDAO.java]: Ok, I got it; you are right; in this case, if it's false, we will not render. Good catch

**Review by idoberko2 — APPROVED** [review]

**Pull request merged**